src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -239,8 +239,14 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 			JSONObject params = getParams(requestHeader.getURI().getEscapedQuery());
 
 			if (shortcutImpl != null) {
+				if (!getOptionsParamApi().isDisableKey() && !getOptionsParamApi().isNoKeyForSafeOps()) {
+					if ( ! this.hasValidKey(requestHeader, params)) {
+						throw new ApiException(ApiException.Type.BAD_API_KEY);
```

Comment:
I dont _think_ so - if there should be a key but isnt then an exception is thrown. Otherwise `msg = shortcutImpl.handleShortcut(msg);` is called. If an exception is thrown then its handled later. Or am I missing something?